src/goto-programs/remove_exceptions.cpp

```diff
@@ -327,6 +312,37 @@ void remove_exceptionst::instrument_throw(
     t_dead->source_location=instr_it->source_location;
     t_dead->function=instr_it->function;
   }
+}
+
+/// instruments each throw with conditional GOTOS to the  corresponding
+/// exception handlers
+void remove_exceptionst::instrument_throw(
+  const goto_functionst::function_mapt::iterator &func_it,
+  const goto_programt::instructionst::iterator &instr_it,
+  const remove_exceptionst::stack_catcht &stack_catch,
+  const std::vector<exprt> &locals)
+{
+  PRECONDITION(instr_it->type==THROW);
+
+  const exprt &exc_expr=
+    uncaught_exceptions_domaint::get_exception_symbol(instr_it->code);
+  bool assertion_error=id2string(
+    uncaught_exceptions_domaint::get_exception_type(exc_expr.type())).
+    find("java.lang.AssertionError")!=std::string::npos;
+
+  // we don't count AssertionError (we couldn't catch it anyway
```

Comment:
I believe this is special because the frontend currently makes a `assert` statement as well as the throw, and therefore will reach path termination before reaching the throw opcode. If we want to change that, we should change Java assertion-coverage to catch AssertionError at the top level rather than placing `ASSERT` GOTO opcodes at each source-language assertion.